### PR TITLE
Feature/graphql

### PR DESCRIPTION
**Pull request by vonElfvin**

### CODE REVIEW

File: src/main/resources/schema.graphqls [hunk 1]
```diff
@@ -0,0 +1,31 @@
+schema {
```
Comment (hedvigoscar) [inline src/main/resources/schema.graphqls]: @Iamsamwhoami @FredrikAreschoug 

I'd like you to check this schema out as well 🙂

File: src/main/resources/schema.graphqls [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    query: Query
+    mutation: Mutation
+}
+
+type Query {
+    chargeEstimation(memberId: String): ChargeEstimation,
+    chargeHistory(memberId: String): [Charge!]!
```
Comment (Iamsamwhoami) [inline src/main/resources/schema.graphqls]: `memberId` should be resolved from the token in the headers, not passed in by the consumer :)

File: src/main/resources/schema.graphqls [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+type Mutation {
+}
+
+type Charge {
+    amount: MonetaryAmountV2!,
+    date: LocalDate!
+}
+
+type ChargeEstimation {
+    currentBalance: MonetaryAmountV2!,
```
Comment (Iamsamwhoami) [inline src/main/resources/schema.graphqls]: As this is not an estimation (i assume) maybe it shouldn't be inside this type?

File: src/main/resources/schema.graphqls [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    estimatedSubscription: MonetaryAmountV2!,
```
Comment (Iamsamwhoami) [inline src/main/resources/schema.graphqls]: it feels a bit redundant to prefix these with `estimated`

File: src/main/kotlin/com/hedvig/accountservice/graphql/extensions/DataFetchingEnviromentExtensions.kt [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package com.hedvig.accountservice.graphql.extensions
+
+import graphql.schema.DataFetchingEnvironment
+import graphql.servlet.GraphQLContext
+
+private const val HEDVIG_TOKEN: String = "hedvig.token"
+
+fun DataFetchingEnvironment.getToken() =
```
Comment (hedvigoscar) [inline src/main/kotlin/com/hedvig/accountservice/graphql/extensions/DataFetchingEnviromentExtensions.kt]: Actually so good

File: src/main/kotlin/com/hedvig/accountservice/graphql/resolvers/Query.kt [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+        val memberId = env.getToken()
+        val currentBalance = accountManager.getCurrentBalance(memberId)
+
+        val estimatedSubscription = getEstimatedSubscription(memberId)
+        val estimatedDiscount = getEstimatedDiscount(memberId)
+
+        val failedCharges = if (currentBalance > estimatedSubscription.subtract(estimatedDiscount)) {
+            val chargeStates = stateManager.findAllChargeStates(memberId)
+            chargeStates
+                .subList(chargeStates.indexOfLast { it.status == ChargeStatus.CHARGE_COMPLETED }.coerceAtLeast(0), chargeStates.size)
```
Comment (vonElfvin) [inline src/main/kotlin/com/hedvig/accountservice/graphql/resolvers/Query.kt]: Does this fallback to index 0 or the last index when no completed charge is available? I.e. if we've never been able to charge a member ever.

File: pom.xml [hunk 7]
```diff
@@ -272,6 +262,35 @@
...
+            <version>5.9.0</version>
+        </dependency>
+        <dependency>
+            <groupId>com.graphql-java-kickstart</groupId>
+            <artifactId>graphiql-spring-boot-starter</artifactId>
+            <version>5.9.0</version>
+        </dependency>
+
+        <!--- GraphQl Dependencies -->
+<!--        <dependency>-->
```
Comment (meletios) [inline pom.xml]: do we need the commented lines?

File: src/main/kotlin/com/hedvig/accountservice/schedulers/model/SchedulerStateRepository.kt [hunk 8]
```diff
@@ -61,4 +61,13 @@ interface SchedulerStateRepository {
...
     fun findAllLatestMemberIdsOfStatus(@Bind status: ChargeStatus): List<String>
+
+    @SqlQuery(
+        """
+            SELECT *
+            FROM "scheduler_states"
+            WHERE member_id=:memberId
+        """
+    )
+    fun findAllChargeStateOfMember(@Bind memberId: String): List<SchedulerState>
```
Comment (meletios) [inline src/main/kotlin/com/hedvig/accountservice/schedulers/model/SchedulerStateRepository.kt]: Should we include all the charges states here? blacklisted? or waiting for approval? 

would it be better to show only     
   CHARGE_REQUESTING,
    CHARGE_REQUEST_FAILED,
    CHARGE_INITIATED,
    CHARGE_FAILED,
    CHARGE_COMPLETED,

File: src/main/kotlin/com/hedvig/accountservice/graphql/resolvers/Mutation.kt [hunk 9]
```diff
@@ -0,0 +1,7 @@
+package com.hedvig.accountservice.graphql.resolvers
+
+import com.coxautodev.graphql.tools.GraphQLMutationResolver
+import org.springframework.stereotype.Component
+
+@Component
+class Mutation: GraphQLMutationResolver
```
Comment (meletios) [inline src/main/kotlin/com/hedvig/accountservice/graphql/resolvers/Mutation.kt]: we can remove this i think since we dont have a mutation

**Pull request merged**